api/src/main/java/org/hawkular/apm/api/services/internal/CommunicationSummaryTreeBuilder.java

```diff
@@ -58,7 +59,11 @@
         List<CommunicationSummaryStatistics> ret = new ArrayList<>();
         for (String endpoint : endpoints) {
             if (nodeMap.containsKey(endpoint)) {
-                CommunicationSummaryStatistics rootNode = new CommunicationSummaryStatistics(nodeMap.get(endpoint));
+                // Check if a 'client' node also exists for the endpoint, and if so, use this as the
+                // initial endpoint
+                CommunicationSummaryStatistics n = nodeMap.get(nodeMap.containsKey(EndpointUtil.encodeClientURI(endpoint)) ?
+                        EndpointUtil.encodeClientURI(endpoint) : endpoint);
+                CommunicationSummaryStatistics rootNode = new CommunicationSummaryStatistics(n);
```

Comment:
If statement (lines 61-70) there are minimally 3 accesses to the collection. This can be rewritten to use min. 1 access and worst 2 accesses. If there is no entry for a key `EndpointUtil.encodeClientURI(endpoint)` use `endpoint` as a key.